src/libbson/src/bson/bson.c

```diff
@@ -3271,16 +3274,18 @@ _bson_as_json_visit_all (const bson_t *bson,
          *length = 3;
       }
 
-      return bson_strdup ("{ }");
+      char *res = is_outermost_array ? "[ ]" : "{ }";
+      return bson_strdup (res);
```

Comment:
```suggestion
      return bson_strdup (is_outermost_array ? "[ ]" : "{ }");
```

Suggested simplification.